cpp/inc/bond/protocol/compact_binary.h

```diff
@@ -503,7 +503,7 @@ class CompactBinaryReader
 };
 
 template <typename Buffer>
-const uint16_t CompactBinaryReader<Buffer>::magic = COMPACT_PROTOCOL;
+BOND_CONSTEXPR_OR_CONST uint16_t CompactBinaryReader<Buffer>::magic;
```

Comment:
Minor: we're using `BufferT` above, but `Buffer` here. Consider `BufferT` here as well.